src/compiler/program.ts

```diff
@@ -2747,6 +2768,16 @@ namespace ts {
             resolvedTypeReferenceDirective?: ResolvedTypeReferenceDirective,
             refFile?: RefFile
         ): void {
+            tracing.begin(tracing.Phase.Program, "processTypeReferenceDirective", { directive: typeReferenceDirective, hasResolved: !!resolveModuleNamesReusingOldState, refKind: refFile?.kind, refPath: refFile?.file.path });

```

Comment:
Shouldnt this be inside resolveTypeReferenceDirectiveNamesWorker instead of  here like resolveModuleNamesWorker ?

Reply:
I added this tracepoint so I could figure out whether one type references was substantially more expensive than the others.  I can also add one in `resolveTypeReferenceDirectiveNamesWorker` though.